syntax = "proto3";

package fraud_detection;

service FraudDetectionService {
    rpc DetectFraud (DetectFraudRequest) returns (DetectFraudResponse);
}

message DetectFraudRequest {
    string userName = 1;
    CreditCard creditCard = 2;
}

message DetectFraudResponse {
    bool isFraud = 1;
}

message CreditCard {
    string number = 1;
    string expirationDate = 2;
    string cvv = 3;
}